### PR TITLE
chore: raise black and isort lint errors in CI

### PR DESCRIPTION
**Pull request by tserg**
### What I did

To raise `black` and `isort` lint errors in CI

### How I did it

WIP

### How to verify it

WIP

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by fubuloubu** [comment]
These work for us: https://github.com/ApeWorX/ape/blob/main/.github/workflows/test.yaml#L6-L44

**Comment by tserg** [comment]
> These work for us: https://github.com/ApeWorX/ape/blob/main/.github/workflows/test.yaml#L6-L44

Is it okay for Github actions to have its own defined workflow instead of invoking `tox.ini`? This would simplify a lot. Currently, I am trying to process the console output after invoking `tox` to check if an error should be raised for the Github action.

**Comment by tserg** [comment]
Update: I have replicated the commands in `tox`, and added the `if: always()` line to ensure that the `flake8` and `isort` tests continue to run even if `black` exits with errors. If there are no other comments, I will squash the commits.

**Comment by fubuloubu** [comment]
> > These work for us: https://github.com/ApeWorX/ape/blob/main/.github/workflows/test.yaml#L6-L44
> 
> Is it okay for Github actions to have its own defined workflow instead of invoking `tox.ini`? This would simplify a lot. Currently, I am trying to process the console output after invoking `tox` to check if an error should be raised for the Github action.

I think it would be possible to just add the same command line flags to `black` and `isort` in the `tox.ini` file

But yes, we could also refactor the Actions workflow files to not use tox. I find it to just be another level of indirection, unnecessary for development purposes when we have CI. Not sure what you think about that @charles-cooper

**Comment by tserg** [comment]
Understood. I have updated Github Actions to be independent of `tox.ini` for the `lint` test. 

Should I also resolve the outstanding lint errors based on the updated CI?

**Comment by charles-cooper** [comment]
> Understood. I have updated Github Actions to be independent of `tox.ini` for the `lint` test.
> 
> Should I also resolve the outstanding lint errors based on the updated CI?

Yes!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,11 +15,19 @@ jobs:
...
+      run: pip install .[lint]
 
-    - name: Run Tox
-      run: TOXENV=lint tox -r
+    - name: Run Black
+      run: black --check --force-exclude=vyper/version.py ./vyper ./tests ./setup.py
+
+    - name: Run flake8
+      run: flake8 ./vyper ./tests
+      if: always()
```
Comment (fubuloubu) [inline .github/workflows/test.yml:26]: what does this do?

**Review by fubuloubu — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (tserg) [reply .github/workflows/test.yml:26]: Currently, the execution order is `black` > `flake8` > `isort`. If `black` flags any errors and raises an exit code, the `lint` test will end, meaning `flake8` and `isort` will not run (for example, see [this run](https://github.com/vyperlang/vyper/runs/4769987813?check_suite_focus=true). The `always()` flag is used to ensure that `flake8` runs regardless whether `black` raises an exit code, and similarly that `isort` runs regardless whether `black` or `isort` raises an exit code.

Comment (fubuloubu) [reply .github/workflows/test.yml:26]: ah neat!

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -15,11 +15,19 @@ jobs:
...
+    - name: Install Dependencies
+      run: pip install .[lint]
 
-    - name: Run Tox
-      run: TOXENV=lint tox -r
+    - name: Run Black
+      run: black --check --force-exclude=vyper/version.py ./vyper ./tests ./setup.py
+
+    - name: Run flake8
+      run: flake8 ./vyper ./tests
```
Comment (charles-cooper) [inline .github/workflows/test.yml:25]: why not also `setup.py`?

**Review by fubuloubu — APPROVED** [review]
Much better than the initial approach of parsing output text

File: .github/workflows/test.yml (hunk 2 above)
Comment (tserg) [reply .github/workflows/test.yml:25]: I was replicating the coverage from `tox.ini` so I am not sure why `setup.py` was originally excluded from `flake8`.

https://github.com/vyperlang/vyper/blob/dd3dcb86ddefe2929bf8f888ededef18ac59f2d4/tox.ini#L46-L52

I did some digging, and noticed this [commit](https://github.com/vyperlang/vyper/commit/9ec8f9de6c7d2aced7b57a0c4cf2528aa6b44c17#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449) where `setup.py` was added to `black` and `isort` but specifically left out for `flake8`. However, it is not explained why `flake8` did not include `setup.py`.

In any event, I ran `flake8` on `setup.py` locally, and only one issue is flagged. 
```
setup.py:56:20: W605 invalid escape sequence '\.'
setup.py:56:25: W605 invalid escape sequence '\d'
```
https://github.com/vyperlang/vyper/blob/dd3dcb86ddefe2929bf8f888ededef18ac59f2d4/setup.py#L56
It seems minor and can be resolved by the following change
```
return re.sub(r"\.dev\d+", "", version_str) 
```

Given the above, should I proceed to add `setup.py` to `flake8`?

Comment (charles-cooper) [reply .github/workflows/test.yml:25]: Yes - makes sense to be the same as the others

Comment (tserg) [reply .github/workflows/test.yml]: Updated!

**Review by fubuloubu — APPROVED** [review]

**Review by charles-cooper — APPROVED** [review]

**Pull request merged**